src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/NamespaceCache.java

```diff
@@ -57,22 +60,80 @@ protected Uni<Boolean> isVectorEnabled(String collectionName) {
+                Boolean vectorEnabled =
+                    table.get().getColumnsList().stream()
+                        .anyMatch(
+                            c ->
+                                c.getName()
+                                    .equals(
+                                        DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME));
+                if (vectorEnabled) {
+                  final Optional<Schema.CqlIndex> vectorIndex =
+                      table.get().getIndexesList().stream()
+                          .filter(
+                              i ->
+                                  i.getColumnName()
+                                      .equals(
+                                          DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME))
+                          .findFirst();
+                  CollectionProperty.SimilarityFunction function =
+                      CollectionProperty.SimilarityFunction.COSINE;
+                  if (vectorIndex.isPresent()) {
+
+                    if (vectorIndex
+                        .get()
+                        .getOptions()
+                        .containsKey(DocumentConstants.Fields.VECTOR_INDEX_FUNCTION_NAME)) {
+                      function =
+                          CollectionProperty.SimilarityFunction.fromString(
+                              vectorIndex
+                                  .get()
+                                  .getOptions()
+                                  .get(DocumentConstants.Fields.VECTOR_INDEX_FUNCTION_NAME));
+                    }
+                  }
+                  return new CollectionProperty(vectorEnabled, function);
+                } else {
+                  return new CollectionProperty(
+                      vectorEnabled, CollectionProperty.SimilarityFunction.UNDEFINED);
+                }
               } else {
                 throw new RuntimeException(
                     ErrorCode.INVALID_COLLECTION_NAME.getMessage() + collectionName);
               }
             });
   }
+
+  public record CollectionProperty(Boolean vectorEnabled, SimilarityFunction similarityFunction) {
+
+    /**
+     * The similarity function used for the vector index. This is only applicable if the vector
+     * index is enabled.
+     */
+    public enum SimilarityFunction {
+      COSINE,
+      EUCLIDEAN,
+      DOT_PRODUCT,
+      UNDEFINED;
+
+      public static SimilarityFunction fromString(String similarityFunction) {
+        if (similarityFunction == null) return UNDEFINED;
+        return switch (similarityFunction) {
+          case "cosine" -> COSINE;
+          case "euclidean" -> EUCLIDEAN;
+          case "dot_product" -> DOT_PRODUCT;
+          default -> throw new JsonApiException(
+              ErrorCode.VECTOR_SEARCH_INVALID_FUCTION_NAME,
+              ErrorCode.VECTOR_SEARCH_INVALID_FUCTION_NAME.getMessage() + similarityFunction);
```

Comment:
Does this need a space between `getMessage()` and `similarityFunction` so we don't end up with `Invalid vector search function namefoo`?

Reply:
Added space as suggested. This error will never be hit because its validated while collection creation,